### PR TITLE
Browser support

### PR DESCRIPTION
**Pull request by NotNite**
- web-preload is now a library
- Hacky build system support to bundle core-extensions by appending them to the top of the file (what the fuck was I thinking tbh)
- Yoinked part of the manifest from Vencord because lol mv3 is so cryptic

TODO:
- [ ] Make installing extensions work
  - oh god asar
- [ ] Test desktop (I haven't at all)
- [ ] Clean up some very no no bad security problems
  - CSP workaround terrifies me

### CODE REVIEW

**Comment by NotNite** [comment]
Updated this PR:
- Merged with develop
- Introduced `webTarget` and `nodeTarget` drop labels to eliminate repeating ourselves depending on if it's web-preload/browser or node-preload/injector
- Made config reading and writing async (it was only called from async places, minus the logger, which now has an "init" function for the log level)
- Made extension loading function with an interface for the filesystem
  - Interface is async but it uses sync fs calls on desktop, we should fix that maybe
  - Basically duplicates the browser FS type but I don't care that much
- Wrote my own .asar parser (iffy on spec compliance)
  - This is used in desktop too, it will need more testing
  - Reading only right now
- Brought in [ZenFS](https://www.npmjs.com/package/@zenfs/core) to act as a virtual filesystem, exposed in `window` so moonbase can get to it
- Tested on desktop (but it needs more testing!) and Chrome (stable, installed an extension, that's about it)

**Review by adryd325 — APPROVED** [review]
